src/Components/Generator/DynamicInputContainer.js

```diff
@@ -216,13 +216,34 @@ export default class DynamicInputContainer extends preact.Component {
     }
 
     removeDynamicInput(event) {
-        if (window.confirm(t('confirm-input-remove', 'generator'))) {
-            this.setState(prev => {
-                delete prev.fields[event.target.getAttribute('rel')];
-                return prev;
-            });
-            this.pushStateUp();
+        // prompt only if respective field has value entered while removing
+        if (!this.checkInputData(event)) {
+            this.confirmStateChange(event);
+        } else if(window.confirm(t('confirm-input-remove', 'generator'))) {
+            this.confirmStateChange(event);
+        }   
+    }
+
+    confirmStateChange(event) {
+        this.setState(prev => {
+            delete prev.fields[event.target.getAttribute('rel')];
+            return prev;
+        });
+        this.pushStateUp();
+    }
+
+    // returns whether the field has value based on its type
+    checkInputData(event) {
```

Comment:
Try hinting to the return value by naming the method in such a way that you can read it easily when it is used in other code, e.g. `isFieldEmpty()` (in this case the return values should obviously be switched).